In [26]:
import pandas as pd
from Univariate import Univariate
function = Univariate

dataset = pd.read_csv("Placement.csv")
dataset

Central tendency

In [5]:
dataset["ssc_p"].mean()

np.float64(67.30339534883721)

In [6]:
dataset["ssc_p"].median()


67.0

In [12]:
dataset["ssc_p"].mode()[0]


np.float64(62.0)

In [13]:
dataset.describe()

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
count,215.000000,215.000000,215.000000,215.000000,215.000000,215.000000,148.000000
mean,108.000000,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
std,62.209324,10.827205,10.897509,7.358743,13.275956,5.833385,93457.452420
min,1.000000,40.890000,37.000000,50.000000,50.000000,51.210000,200000.000000
25%,54.500000,60.600000,60.900000,61.000000,60.000000,57.945000,240000.000000
50%,108.000000,67.000000,65.000000,66.000000,71.000000,62.000000,265000.000000
75%,161.500000,75.700000,73.000000,72.000000,83.500000,66.255000,300000.000000
max,215.000000,89.400000,97.700000,91.000000,98.000000,77.890000,940000.000000


In [14]:
quan, qual = function.quanQual(dataset)
print(quan)
print(qual)



In [29]:
descresptive = pd.DataFrame()

In [35]:
descresptive = pd.DataFrame(index=["Mean", "Median", "Mode"], columns=quan)
for columnName in quan:
    descresptive[columnName]["Mean"] = dataset[columnName].mean()
    descresptive[columnName]["Median"] = dataset[columnName].median()
    descresptive[columnName]["Mode"] = dataset[columnName].mode()[0]
    

/var/folders/xh/qbyhzlq91t96qsm5mpr0mrqr0000gn/T/ipykernel_1333/923913756.py:3: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

  descresptive[columnName]["Mean"] = dataset[columnName].mean()
/var/folders/xh/qbyhzlq91t96qsm5mpr0mrqr0000gn/T/ipyk

In [36]:
descresptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0


The dataset results for mean, median, and mode are as follows:
	Mean overall percentages:
SSC_P = 67.30, HSC_P = 66.33, Degree_P = 66.37, ETest_P = 72.10, MBA_P = 62.27.
The corresponding average employee salary is ₹288,655.40.
	Median overall percentages:
SSC_P = 67.0, HSC_P = 65.0, Degree_P = 66.0, ETest_P = 71.0, MBA_P = 62.0.
The corresponding median employee salary is ₹265,000.00.
	Mode overall percentages:
SSC_P = 62.0, HSC_P = 63.0, Degree_P = 65.0, ETest_P = 60.0, MBA_P = 56.7.
The corresponding mode employee salary is ₹300,000.00.